modules/codegen/src/smithy4s/codegen/transformers/AwsConstraintsRemover.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.codegen.transformers
+
+import software.amazon.smithy.build.ProjectionTransformer
+import software.amazon.smithy.build.TransformContext
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.neighbor.NeighborProvider
+import scala.jdk.CollectionConverters._
+
+/** A preprocessor that removes constraints from specifications of AWS. */
+class AwsConstraintsRemover extends ProjectionTransformer {
+
+  override def getName: String = AwsConstraintsRemover.name
+
+  private val constraintsToRemove = List("length", "range", "pattern")
+  private val awsNamespacePrefix = "com.amazonaws"
+
+  override def transform(context: TransformContext): Model = {
+    val transformer = context.getTransformer()
+    transformer.removeTraitsIf(
```

Comment:
Okay, so you need to build a collection of all shapes connected to AWS outputs, before calling the `removeTraitsIf`. I've not tested it, but looks like this : 

```scala
    val model = context.getModel()
    val walker: Walker = new Walker(
      NeighborProvider.cached(NeighborProvider.of(context.getModel()))
    )
    val awsOperations = model
      .getOperationShapes()
      .asScala
      .filter(_.getId().getNamespace().startsWith(awsNamespacePrefix))

    val awsOutputShapes =
      awsOperations.flatMap(_.getInput().toScala.toSet).map(model.expectShape)
    val awsOutputConnectedShapes: Set[ShapeId] =
      awsOutputShapes.flatMap(walker.walkShapeIds(_).asScala)

```

The condition for removing the trait is basically a check to verify whether the shape's ID is in the `awsOutputConnectedShape` set.